staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/cbor.go

```diff
@@ -0,0 +1,29 @@
+package v1
+
+import (
+	"errors"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+func IsMapOrNull(in []byte) bool {
+	decMode, err := cbor.DecOptions{
+		DupMapKey:           cbor.DupMapKeyQuiet,
+		FieldNameMatching:   cbor.FieldNameMatchingCaseSensitive,
+		FieldNameByteString: cbor.FieldNameByteStringAllowed,
+	}.DecMode()
+	if err != nil {
+		panic(err)
+	}
+
+	dst := new(struct{})
+	if err := decMode.Unmarshal(in, &dst); err != nil {
+		var uterr *cbor.UnmarshalTypeError
+		if errors.As(err, &uterr) && uterr.GoType == "string" {
+			return true
```

Comment:
Worth a comment here. This essentially means that the input was a map but one of its keys was not a string type, I think.